homeassistant/components/elmo_alarm/__init__.py

```diff
@@ -0,0 +1,232 @@
+    username = conf[CONF_USERNAME]
+    password = conf[CONF_PASSWORD]
+    states = conf[CONF_STATES]
+    scan_interval = conf[CONF_SCAN_INTERVAL]
+
+    _LOGGER.warning(f"Scan Interval:{scan_interval}")
+
+    client = ElmoClientWrapper(host, vendor, username, password, states)
+    await client.update()
+
+    hass.data[DOMAIN] = client
+
+    hass.async_create_task(
+        async_load_platform(
+            hass,
+            "binary_sensor",
+            DOMAIN,
+            {"zones": client._zones, "inputs": client._inputs},
+            config,
+        )
+    )
+
+    hass.async_create_task(
+        async_load_platform(hass, "alarm_control_panel", DOMAIN, {}, config)
+    )
+
+    async def update():
+        _LOGGER.debug("Connecting to e-connect to retrieve states.")
+        await client.update()
+        async_dispatcher_send(hass, SIGNAL_ARMING_STATE_CHANGED, client._state)
+
+        for zone in client._zones:
+            async_dispatcher_send(hass, SIGNAL_ZONE_CHANGED, zone)
+
+        for inp in client._inputs:
+            async_dispatcher_send(hass, SIGNAL_INPUT_CHANGED, inp)
+
+        async_call_later(
+            hass,
+            scan_interval.total_seconds(),
+            lambda _: hass.async_create_task(update()),
+        )
+
+    await update()
+
+    return True
+
+
+class ElmoClientWrapper(ElmoClient):
+    """Wrapping the Elmo client class to adapt it to home assistant"""
+
+    def __init__(self, host, vendor, username, password, states_config):
+        self._username = username
+        self._password = password
+        self._states_config = states_config
+        self._data = None
+        self._states = None
+        self._state = None
+
+        ElmoClient.__init__(self, host, vendor)
+
+    async def update(self):
+        """Get updates and refresh internal states"""
+        try:
+            data = self.check()
+        except PermissionDenied as e:
+            _LOGGER.warning(f"Invalid session, trying to authenticate: {e}.")
+            try:
+                self.auth(self._username, self._password)
+                data = self.check()
+            except PermissionDenied as e:
+                _LOGGER.warning(f"Invalid credentials: {e}.")
+            except HTTPError as e:
+                _LOGGER.warning(
+                    f"Got HTTP error when authenticating. Check credentials. Code: {e.response.status_code}."
+                )
+
+        if self._data is None:
+            await self._configure_states()
+        self._data = data
+        await self._update_arm_state()
+        await self._update_zone_state()
+        await self._update_input_state()
+
+    async def _configure_states(self):
+        state_config = {}
+        for state in self._states_config:
+            state_config[state[CONF_NAME]] = state[CONF_ZONES]
+        self._states = state_config
+
+    async def _update_arm_state(self):
+        areas_armed = self._data["areas_armed"]
+
+        if len(areas_armed) == 0:
```

Comment:
Use it as a falsy value:
```suggestion
        if not areas_armed:
```

Reply:
Done.